deployment/keystone/changeset/operations/contracts/deploy_contracts_seq.go

```diff
@@ -104,14 +107,18 @@ var DeployKeystoneContractsSequence = operations.NewSequence[DeployKeystoneContr
 			}
 		}
 		if input.DeployEVMOCR3 {
-			// EVM cap OCR3 Contract
-			evmOCR3DeployReport, err := operations.ExecuteOperation(b, DeployOCR3Op, DeployOCR3OpDeps(deps), DeployOCR3OpInput{ChainSelector: input.RegistryChainSelector, Qualifier: "capability_evm"})
-			if err != nil {
-				return DeployKeystoneContractsSequenceOutput{}, err
-			}
-			err = updateAddresses(as.Addresses(), evmOCR3DeployReport.Output.Addresses, ab, evmOCR3DeployReport.Output.AddressBook)
-			if err != nil {
-				return DeployKeystoneContractsSequenceOutput{}, err
+			for chainID, selector := range input.EVMChainIDs {
+				// EVM cap OCR3 Contract
+				fmt.Printf("Processing EVM Block Chain ID: %d, selector: %d\n", chainID, selector)
+				qualifier := fmt.Sprintf("capability_evm_%d", chainID)
+				evmOCR3DeployReport, err := operations.ExecuteOperation(b, DeployOCR3Op, DeployOCR3OpDeps(deps), DeployOCR3OpInput{ChainSelector: selector, Qualifier: qualifier})
```

Comment:
hmm... now... I do wonder about one thing: should all OCR3 contracts be deployed on the home chain (and only configured to work with other chains) or should they be deployed on the chains, with which that capability will interact? I see you did the latter, but are we sure it should work like that?

Reply:
with the former option:
> should all OCR3 contracts be deployed on the home chain (and only configured to work with other chains) 
I had issues when deploying across two chains: the OCR rounds on the EVM capability side did not reach consensus, and the data didn't flow back to the workflow. But it could be a configuration issue.

That's why I defaulted in deploying a contract per chain, and then properly select it in the job spec [here](https://github.com/smartcontractkit/chainlink/pull/19008/files#diff-eea75e016d9ab8a591e01785f3a198ebd9aa46c90a926a5fcb6ae00ab03f6600R55) by using the right chain selector (and not the registry chain/home chain)

Can we just merge it as is and debug this later on if required, or do you believe Bartek this needs more digging?